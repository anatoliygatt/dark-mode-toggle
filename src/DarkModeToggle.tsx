import * as React from 'react';
import { useIsTransitionDisabled } from './hooks/useIsTransitionDisabled';
import { TOGGLE_TRANSITION_DURATION, StyledDarkModeToggle } from './style';

export type Mode = 'dark' | 'light';

export type Size = 'sm' | 'md' | 'lg';

export interface StyledProps {
  size?: Size;
  inactiveLabelColor?: string;
  inactiveLabelColorOnHover?: string;
  inactiveLabelColorOnActive?: string;
  activeLabelColor?: string;
  activeLabelColorOnHover?: string;
  activeLabelColorOnActive?: string;
  inactiveTrackColor?: string;
  inactiveTrackColorOnHover?: string;
  inactiveTrackColorOnActive?: string;
  activeTrackColor?: string;
  activeTrackColorOnHover?: string;
  activeTrackColorOnActive?: string;
  inactiveThumbColor?: string;
  activeThumbColor?: string;
  focusRingColor?: string;
}

export interface AccessibilityProps {
  ariaLabel?: string;
}

export interface Props extends StyledProps, AccessibilityProps {
  mode?: Mode;
  dark?: string;
  light?: string;
  onChange?: (mode: Mode) => void;
}

export function DarkModeToggle({
  mode = 'dark',
  dark,
  light,
  onChange,
  size = 'sm',
  inactiveLabelColor = '#b9bdc1',
  inactiveLabelColorOnHover = '#fcfefe',
  inactiveLabelColorOnActive = '#cdd1d5',
  activeLabelColor = '#5b5e62',
  activeLabelColorOnHover = '#404346',
  activeLabelColorOnActive = '#010101',
  inactiveTrackColor = '#dce0e3',
  inactiveTrackColorOnHover = '#fcfefe',
  inactiveTrackColorOnActive = '#cdd1d5',
  activeTrackColor = '#404346',
  activeTrackColorOnHover = '#2d2f31',
  activeTrackColorOnActive = '#141516',
  inactiveThumbColor = '#2d2f31',
  activeThumbColor = '#dce0e3',
  focusRingColor = 'rgb(59 130 246 / 0.5)',
  ariaLabel,
}: Props) {
  const isTransitionDisabled = useIsTransitionDisabled(
    size,
    TOGGLE_TRANSITION_DURATION
  );

  return (
    <StyledDarkModeToggle
      size={size}
      inactiveLabelColor={inactiveLabelColor}
      inactiveLabelColorOnHover={inactiveLabelColorOnHover}
      inactiveLabelColorOnActive={inactiveLabelColorOnActive}
      activeLabelColor={activeLabelColor}
      activeLabelColorOnHover={activeLabelColorOnHover}
      activeLabelColorOnActive={activeLabelColorOnActive}
      inactiveTrackColor={inactiveTrackColor}
      inactiveTrackColorOnHover={inactiveTrackColorOnHover}
      inactiveTrackColorOnActive={inactiveTrackColorOnActive}
      activeTrackColor={activeTrackColor}
      activeTrackColorOnHover={activeTrackColorOnHover}
      activeTrackColorOnActive={activeTrackColorOnActive}
      inactiveThumbColor={inactiveThumbColor}
      activeThumbColor={activeThumbColor}
      focusRingColor={focusRingColor}
      data-transition-disabled={isTransitionDisabled}
      data-checked={mode === 'light'}
      data-testid="dark-mode-toggle"
    >
      {dark && (
        <span
          onClick={() => {
            onChange?.('dark');
          }}
          aria-hidden="true"
        >
          {dark}
        </span>
      )}
      <input
        type="checkbox"
        role="switch"
        checked={mode === 'light'}
        aria-label={ariaLabel}
        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
          onChange?.(event.target.checked ? 'light' : 'dark');
        }}
      />
      {light && (
        <span
          onClick={() => {
            onChange?.('light');
          }}
          aria-hidden="true"
        >
          {light}
        </span>
      )}
    </StyledDarkModeToggle>
  );
}
