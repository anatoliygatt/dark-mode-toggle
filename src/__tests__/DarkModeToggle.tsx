import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';
import { DarkModeToggle } from '../DarkModeToggle';
import type { Mode } from '../DarkModeToggle';

jest.mock('../hooks/useIsTransitionDisabled', () => {
  return {
    useIsTransitionDisabled: jest.fn(() => false),
  };
});

describe('DarkModeToggle', () => {
  test('renders without crashing', () => {
    expect(() => {
      render(<DarkModeToggle />);
    }).not.toThrowError();
  });

  test('renders into the document', () => {
    render(<DarkModeToggle />);
    expect(screen.getByTestId('dark-mode-toggle')).toBeInTheDocument();
  });

  test('renders with the default size and colors', () => {
    render(<DarkModeToggle />);
    expect(screen.getByTestId('dark-mode-toggle')).toMatchSnapshot();
  });

  test('renders with the size="md" and default colors', () => {
    render(<DarkModeToggle size="md" />);
    expect(screen.getByTestId('dark-mode-toggle')).toMatchSnapshot();
  });

  test('renders with the size="lg" and custom colors', () => {
    render(
      <DarkModeToggle
        size="lg"
        inactiveTrackColor="#e2e8f0"
        inactiveTrackColorOnHover="#f8fafc"
        inactiveTrackColorOnActive="#cbd5e1"
        activeTrackColor="#334155"
        activeTrackColorOnHover="#1e293b"
        activeTrackColorOnActive="#0f172a"
        inactiveThumbColor="#1e293b"
        activeThumbColor="#e2e8f0"
      />
    );
    expect(screen.getByTestId('dark-mode-toggle')).toMatchSnapshot();
  });

  test('renders with the labels', () => {
    render(<DarkModeToggle dark="Dark" light="Light" />);
    expect(screen.getByTestId('dark-mode-toggle')).toMatchSnapshot();
  });

  test('renders with an aria-label attribute when ariaLabel="Toggle color scheme"', () => {
    render(<DarkModeToggle ariaLabel="Toggle color scheme" />);
    expect(screen.getByRole('switch')).toHaveAttribute(
      'aria-label',
      'Toggle color scheme'
    );
  });

  test('responds to a user interaction', async () => {
    function ControlledDarkModeToggle() {
      const [mode, setMode] = React.useState<Mode>('dark');
      return (
        <DarkModeToggle
          dark="Dark"
          light="Light"
          mode={mode}
          onChange={(mode) => {
            setMode(mode);
          }}
        />
      );
    }

    render(<ControlledDarkModeToggle />);

    expect(screen.getByRole('switch')).not.toBeChecked();
    await userEvent.click(screen.getByRole('switch'));
    expect(screen.getByRole('switch')).toBeChecked();
    await userEvent.click(screen.getByRole('switch'));
    expect(screen.getByRole('switch')).not.toBeChecked();

    await userEvent.click(screen.getByText('Light'));
    expect(screen.getByRole('switch')).toBeChecked();
    await userEvent.click(screen.getByText('Dark'));
    expect(screen.getByRole('switch')).not.toBeChecked();
  });

  test('responds to a change in focused state when tabbing with the keyboard', async () => {
    render(<DarkModeToggle />);

    expect(screen.getByRole('switch')).not.toHaveFocus();
    await userEvent.tab();
    expect(screen.getByRole('switch')).toHaveFocus();
    await userEvent.tab();
    expect(screen.getByRole('switch')).not.toHaveFocus();
  });

  test('complies with the web accessibility standards', async () => {
    const { container } = render(
      <DarkModeToggle ariaLabel="Toggle color scheme" />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
