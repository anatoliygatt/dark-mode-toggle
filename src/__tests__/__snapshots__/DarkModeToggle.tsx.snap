// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeToggle renders with the default size and colors 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

.emotion-0 span {
  -webkit-transition: color 250ms ease-in-out;
  transition: color 250ms ease-in-out;
  padding: 8px 12px;
  cursor: pointer;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
}

.emotion-0 span:nth-of-type(1):hover {
  color: #404346;
}

.emotion-0 span:nth-of-type(1):active {
  color: #010101;
}

.emotion-0 span:nth-of-type(2):hover {
  color: #fcfefe;
}

.emotion-0 span:nth-of-type(2):active {
  color: #cdd1d5;
}

.emotion-0 input {
  position: relative;
  overflow: hidden;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: box-shadow,background-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  outline: none;
  border-radius: 14px;
  width: 48px;
  height: 28px;
  background-color: #dce0e3;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 input:checked {
  background-color: #404346;
}

.emotion-0 input:checked:hover,
.emotion-0 input:checked:hover::after {
  background-color: #2d2f31;
}

.emotion-0 input:checked:active,
.emotion-0 input:checked:active::after {
  background-color: #141516;
}

.emotion-0 input:checked::before {
  left: 24px;
  background-color: #dce0e3;
}

.emotion-0 input:checked::after {
  top: 12px;
  right: 2px;
  width: 1px;
  height: 1px;
  border-radius: 0.5px;
  background-color: #404346;
}

.emotion-0 input:hover,
.emotion-0 input:hover::after {
  background-color: #fcfefe;
}

.emotion-0 input:focus,
.emotion-0 input:focus-visible {
  box-shadow: 0 0 0 2px rgb(59 130 246 / 0.5);
}

.emotion-0 input:focus:not(:focus-visible) {
  box-shadow: none;
}

.emotion-0 input:active,
.emotion-0 input:active::after {
  background-color: #cdd1d5;
}

.emotion-0 input::before {
  content: '';
  position: absolute;
  top: 4px;
  left: 4px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: left,background-color;
  width: 20px;
  height: 20px;
  border-radius: 12px;
  background-color: #2d2f31;
}

.emotion-0 input::after {
  content: '';
  position: absolute;
  top: -2px;
  right: 2px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: top,right,width,height,border-radius,background-color;
  width: 32px;
  height: 32px;
  border-radius: 16px;
  background-color: #dce0e3;
}

.emotion-0[data-checked='false'] span {
  color: #b9bdc1;
}

.emotion-0[data-checked='false'] span:nth-of-type(1) {
  pointer-events: none;
}

.emotion-0[data-checked='true'] span {
  color: #5b5e62;
}

.emotion-0[data-checked='true'] span:nth-of-type(2) {
  pointer-events: none;
}

.emotion-0[data-transition-disabled='true'] *,
.emotion-0[data-transition-disabled='true'] *::before,
.emotion-0[data-transition-disabled='true'] *::after {
  transition-duration: 0ms;
}

<div
  class="emotion-0"
  data-checked="false"
  data-testid="dark-mode-toggle"
  data-transition-disabled="false"
>
  <input
    role="switch"
    type="checkbox"
  />
</div>
`;

exports[`DarkModeToggle renders with the labels 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

.emotion-0 span {
  -webkit-transition: color 250ms ease-in-out;
  transition: color 250ms ease-in-out;
  padding: 8px 12px;
  cursor: pointer;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
}

.emotion-0 span:nth-of-type(1):hover {
  color: #404346;
}

.emotion-0 span:nth-of-type(1):active {
  color: #010101;
}

.emotion-0 span:nth-of-type(2):hover {
  color: #fcfefe;
}

.emotion-0 span:nth-of-type(2):active {
  color: #cdd1d5;
}

.emotion-0 input {
  position: relative;
  overflow: hidden;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: box-shadow,background-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  outline: none;
  border-radius: 14px;
  width: 48px;
  height: 28px;
  background-color: #dce0e3;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 input:checked {
  background-color: #404346;
}

.emotion-0 input:checked:hover,
.emotion-0 input:checked:hover::after {
  background-color: #2d2f31;
}

.emotion-0 input:checked:active,
.emotion-0 input:checked:active::after {
  background-color: #141516;
}

.emotion-0 input:checked::before {
  left: 24px;
  background-color: #dce0e3;
}

.emotion-0 input:checked::after {
  top: 12px;
  right: 2px;
  width: 1px;
  height: 1px;
  border-radius: 0.5px;
  background-color: #404346;
}

.emotion-0 input:hover,
.emotion-0 input:hover::after {
  background-color: #fcfefe;
}

.emotion-0 input:focus,
.emotion-0 input:focus-visible {
  box-shadow: 0 0 0 2px rgb(59 130 246 / 0.5);
}

.emotion-0 input:focus:not(:focus-visible) {
  box-shadow: none;
}

.emotion-0 input:active,
.emotion-0 input:active::after {
  background-color: #cdd1d5;
}

.emotion-0 input::before {
  content: '';
  position: absolute;
  top: 4px;
  left: 4px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: left,background-color;
  width: 20px;
  height: 20px;
  border-radius: 12px;
  background-color: #2d2f31;
}

.emotion-0 input::after {
  content: '';
  position: absolute;
  top: -2px;
  right: 2px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: top,right,width,height,border-radius,background-color;
  width: 32px;
  height: 32px;
  border-radius: 16px;
  background-color: #dce0e3;
}

.emotion-0[data-checked='false'] span {
  color: #b9bdc1;
}

.emotion-0[data-checked='false'] span:nth-of-type(1) {
  pointer-events: none;
}

.emotion-0[data-checked='true'] span {
  color: #5b5e62;
}

.emotion-0[data-checked='true'] span:nth-of-type(2) {
  pointer-events: none;
}

.emotion-0[data-transition-disabled='true'] *,
.emotion-0[data-transition-disabled='true'] *::before,
.emotion-0[data-transition-disabled='true'] *::after {
  transition-duration: 0ms;
}

<div
  class="emotion-0"
  data-checked="false"
  data-testid="dark-mode-toggle"
  data-transition-disabled="false"
>
  <span
    aria-hidden="true"
  >
    Dark
  </span>
  <input
    role="switch"
    type="checkbox"
  />
  <span
    aria-hidden="true"
  >
    Light
  </span>
</div>
`;

exports[`DarkModeToggle renders with the size="lg" and custom colors 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

.emotion-0 span {
  -webkit-transition: color 250ms ease-in-out;
  transition: color 250ms ease-in-out;
  padding: 16px 24px;
  cursor: pointer;
  font-size: 24px;
  font-weight: 600;
  text-transform: uppercase;
}

.emotion-0 span:nth-of-type(1):hover {
  color: #404346;
}

.emotion-0 span:nth-of-type(1):active {
  color: #010101;
}

.emotion-0 span:nth-of-type(2):hover {
  color: #fcfefe;
}

.emotion-0 span:nth-of-type(2):active {
  color: #cdd1d5;
}

.emotion-0 input {
  position: relative;
  overflow: hidden;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: box-shadow,background-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  outline: none;
  border-radius: 28px;
  width: 96px;
  height: 56px;
  background-color: #e2e8f0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 input:checked {
  background-color: #334155;
}

.emotion-0 input:checked:hover,
.emotion-0 input:checked:hover::after {
  background-color: #1e293b;
}

.emotion-0 input:checked:active,
.emotion-0 input:checked:active::after {
  background-color: #0f172a;
}

.emotion-0 input:checked::before {
  left: 48px;
  background-color: #e2e8f0;
}

.emotion-0 input:checked::after {
  top: 24px;
  right: 4px;
  width: 2px;
  height: 2px;
  border-radius: 1px;
  background-color: #334155;
}

.emotion-0 input:hover,
.emotion-0 input:hover::after {
  background-color: #f8fafc;
}

.emotion-0 input:focus,
.emotion-0 input:focus-visible {
  box-shadow: 0 0 0 4px rgb(59 130 246 / 0.5);
}

.emotion-0 input:focus:not(:focus-visible) {
  box-shadow: none;
}

.emotion-0 input:active,
.emotion-0 input:active::after {
  background-color: #cbd5e1;
}

.emotion-0 input::before {
  content: '';
  position: absolute;
  top: 8px;
  left: 8px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: left,background-color;
  width: 40px;
  height: 40px;
  border-radius: 24px;
  background-color: #1e293b;
}

.emotion-0 input::after {
  content: '';
  position: absolute;
  top: -4px;
  right: 4px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: top,right,width,height,border-radius,background-color;
  width: 64px;
  height: 64px;
  border-radius: 32px;
  background-color: #e2e8f0;
}

.emotion-0[data-checked='false'] span {
  color: #b9bdc1;
}

.emotion-0[data-checked='false'] span:nth-of-type(1) {
  pointer-events: none;
}

.emotion-0[data-checked='true'] span {
  color: #5b5e62;
}

.emotion-0[data-checked='true'] span:nth-of-type(2) {
  pointer-events: none;
}

.emotion-0[data-transition-disabled='true'] *,
.emotion-0[data-transition-disabled='true'] *::before,
.emotion-0[data-transition-disabled='true'] *::after {
  transition-duration: 0ms;
}

<div
  class="emotion-0"
  data-checked="false"
  data-testid="dark-mode-toggle"
  data-transition-disabled="false"
>
  <input
    role="switch"
    type="checkbox"
  />
</div>
`;

exports[`DarkModeToggle renders with the size="md" and default colors 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

.emotion-0 span {
  -webkit-transition: color 250ms ease-in-out;
  transition: color 250ms ease-in-out;
  padding: 12px 18px;
  cursor: pointer;
  font-size: 18px;
  font-weight: 600;
  text-transform: uppercase;
}

.emotion-0 span:nth-of-type(1):hover {
  color: #404346;
}

.emotion-0 span:nth-of-type(1):active {
  color: #010101;
}

.emotion-0 span:nth-of-type(2):hover {
  color: #fcfefe;
}

.emotion-0 span:nth-of-type(2):active {
  color: #cdd1d5;
}

.emotion-0 input {
  position: relative;
  overflow: hidden;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: box-shadow,background-color;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  outline: none;
  border-radius: 21px;
  width: 72px;
  height: 42px;
  background-color: #dce0e3;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 input:checked {
  background-color: #404346;
}

.emotion-0 input:checked:hover,
.emotion-0 input:checked:hover::after {
  background-color: #2d2f31;
}

.emotion-0 input:checked:active,
.emotion-0 input:checked:active::after {
  background-color: #141516;
}

.emotion-0 input:checked::before {
  left: 36px;
  background-color: #dce0e3;
}

.emotion-0 input:checked::after {
  top: 18px;
  right: 3px;
  width: 1.5px;
  height: 1.5px;
  border-radius: 0.75px;
  background-color: #404346;
}

.emotion-0 input:hover,
.emotion-0 input:hover::after {
  background-color: #fcfefe;
}

.emotion-0 input:focus,
.emotion-0 input:focus-visible {
  box-shadow: 0 0 0 3px rgb(59 130 246 / 0.5);
}

.emotion-0 input:focus:not(:focus-visible) {
  box-shadow: none;
}

.emotion-0 input:active,
.emotion-0 input:active::after {
  background-color: #cdd1d5;
}

.emotion-0 input::before {
  content: '';
  position: absolute;
  top: 6px;
  left: 6px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: left,background-color;
  width: 30px;
  height: 30px;
  border-radius: 18px;
  background-color: #2d2f31;
}

.emotion-0 input::after {
  content: '';
  position: absolute;
  top: -3px;
  right: 3px;
  display: block;
  -webkit-transition: all 250ms ease-in-out;
  transition: all 250ms ease-in-out;
  transition-property: top,right,width,height,border-radius,background-color;
  width: 48px;
  height: 48px;
  border-radius: 24px;
  background-color: #dce0e3;
}

.emotion-0[data-checked='false'] span {
  color: #b9bdc1;
}

.emotion-0[data-checked='false'] span:nth-of-type(1) {
  pointer-events: none;
}

.emotion-0[data-checked='true'] span {
  color: #5b5e62;
}

.emotion-0[data-checked='true'] span:nth-of-type(2) {
  pointer-events: none;
}

.emotion-0[data-transition-disabled='true'] *,
.emotion-0[data-transition-disabled='true'] *::before,
.emotion-0[data-transition-disabled='true'] *::after {
  transition-duration: 0ms;
}

<div
  class="emotion-0"
  data-checked="false"
  data-testid="dark-mode-toggle"
  data-transition-disabled="false"
>
  <input
    role="switch"
    type="checkbox"
  />
</div>
`;
